ipset_test.go

```diff
@@ -236,6 +237,77 @@ func TestIPSetDifference(t *testing.T) {
 	assert.False(t, set.ContainsNet(cidr))
 }
 
+func TestIPSetIntersection(t *testing.T) {
+	testCases := []struct {
+		case1        []string
+		case2        []string
+		intersection []string
+	}{
+		{
+			[]string{"10.0.20.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.0.20.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.23.224.104/27", "10.0.20.2/30", "10.5.8.200/29"},
+		},
+		{
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.10.0.0/20", "10.5.8.0/24", "10.23.224.100/23"},
+			[]string{"10.10.0.2/30", "10.5.8.200/29", "10.23.224.104/27"},
+		},
+		{
+			[]string{"10.0.5.0/24", "10.5.8.200/29", "10.23.224.104/27"},
+			[]string{"10.6.0.0/24", "10.9.9.1/29", "10.23.6.100/23"},
+			[]string{},
+		},
+		{
+			[]string{"10.23.6.100/24", "10.5.8.200/29", "10.23.224.104/27"},
```

Comment:
10.23.6.100/24 isn't valid. You might consider putting up another PR to make this an error.